### ISSUE TITLE
Broken since last update

### ISSUE THREAD

**Issue by garygreen** [expandtheroom/jquery-infinite-scroll-helper#19]
The latest way of getting the scroll container has broken for me. It get's the second page but then doesn't request any further pages. Previous version 1.2.1 worked fine

**Comment by garygreen** [comment]
Ah this is because on one of our parent elements we have `overflow-x: hidden;` and it's determining that to be the scrollable element. Look at this jsfiddle, if you set that css property the `$('div').css('overflow');` value seems to be 'auto' for some reason. That's why it's not working properly. I'm not sure relying on overflow stuff is the best option for working out the scroll container... 

https://jsfiddle.net/vLthrp2a/

**Comment by garygreen** [comment]
```javascript
	/**
	 * Determines if the provided $el is scrollable or not
	 * @param {jQuery} $el The jQuery instance to check
	 * @returns {boolean} Returns true if scrollable, false if not
	 * @private
	 */
	Plugin.prototype._isScrollableElement = function($el) {
		return (/(auto|scroll)/).test($el.css('overflow') + $el.css('overflow-y'));
	};
```

This should just be

```javascript
	Plugin.prototype._isScrollableElement = function($el) {
		return (/(auto|scroll)/).test($el.css('overflow-y'));
	};
```

As if you do `overflow: scroll` the value of `overflow-y` will be scroll too.

**Comment by rogden** [comment]
Hi Gary,

Thanks for the insight. Looking into it now. Stay tuned.

**Comment by rogden** [comment]
@garygreen It seems that setting overflow-x: hidden still creates a y-scrollable container. See example: http://jsbin.com/napemucede/edit?html,css,output

So in your example, `$el.css('overflow-y')` would still return auto. I suppose we could add `&& $el.css('overflow-x') != 'hidden' to the return statement. Though this could potentially backfire if someone sets an element to overflow-y: auto and overflow-x: hidden.

I think the best approach might be to add an option, `scrollContainer`, that allows you to specify the container that the scroll event will be attached to. This would remove any 'guessing' the plugin would have to do in order to figure out which parent element is the scroll container. 

In order to keep the backwards compatibility, setting this option would just short circuit the _getScrollContainer implementation and return the element that the scrollContainer option specified. 

In your use case that is currently breaking with 1.2.2, is the window the desired scrollable element?

**Comment by garygreen** [comment]
Hmm yeah it seems as soon as you set overflow-x it automatically adds scrollbar to the Y axis. Kinda strange.

In any case we didn't actually need the overflow-x on our element, so it's not a problem for us any longer. But +1 to:

> I think the best approach might be to add an option, scrollContainer, that allows you to specify the container that the scroll event will be attached to

**Comment by garygreen** [comment]
@rogden any news on adding option for setting scroll container manually?

**Comment by garygreen** [comment]
We did actually need overflow-x: hidden in the end. Basically we've got a sidebar menu which slides in and so need the x-axis overflow hidden. Infinite scrollbar helper is picking up on that being the scroll container (for reasons above) which it shouldn't be. We're got around this by overloading the `isScrollableElement` for now:

```javascript
		// Fix for: https://github.com/expandtheroom/jquery-infinite-scroll-helper/issues/19
		window.InfiniteScrollHelper.prototype._isScrollableElement = function($el) {
			if ($el.css('overflow-x') == 'hidden') return false;
			return (/(auto|scroll)/).test($el.css('overflow') + $el.css('overflow-y'));
		};
```

Not sure the best way to fix tbh. Maybe specifying a custom scroll container in the options would be most flexible and take a lot of the guess out, as you mentioned.
